app/database/operator/index.ts

```diff
@@ -7,18 +7,18 @@ import GroupHandler, {GroupHandlerMix} from '@database/operator/handlers/group';
 import PostHandler, {PostHandlerMix} from '@database/operator/handlers/post';
 import TeamHandler, {TeamHandlerMix} from '@database/operator/handlers/team';
 import UserHandler, {UserHandlerMix} from '@database/operator/handlers/user';
+import {Database} from '@nozbe/watermelondb';
 import mix from '@utils/mix';
 
 interface Operator extends BaseHandlerMix, PostHandlerMix, UserHandlerMix, GroupHandlerMix, ChannelHandlerMix, TeamHandlerMix {}
 
-class Operator extends mix(BaseHandler).with(
-    PostHandler,
-    UserHandler,
-    GroupHandler,
-    ChannelHandler,
-    TeamHandler,
-) {}
+class Operator extends mix(BaseHandler).with(PostHandler, UserHandler, GroupHandler, ChannelHandler, TeamHandler) {
+    database!: Database;
 
-const DataOperator = new Operator();
+    constructor(database: Database) {
+        super();
+        this.activeDatabase = database;
```

Comment:
> Could we call this class DatabaseOperator?

The purpose of this class is to treat `data` getting inside our database.  It was initially called `DataOperator` and then we agreed to use the term `Operator`.   I would prefer if we leave it as such.  @enahum  your thoughts on this ?


> And also, since we'll have an instance of this class for each database, can we rename this.activeDatabase to this.database?

We are not using Singletons anymore.   The Operator treats/manages data getting inside a server database.  It is the Operator that relies on the DatabaseManager...not the other way around.

 An instance of an operator needs an instance of a database connection for the `activeDatabase` but  every Operator instances will point to the same `Default` database.